### PR TITLE
[SPARK-49920][INFRA] Install `R` for `ubuntu 24.04` when GA run `k8s-integration-tests`

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to install `R` for `ubuntu 24.04` when GA run `k8s-integration-tests`.

### Why are the changes needed?
- As the GitHub community switches the default version of `ubuntu-latest` from `ubuntu-22.04` to `ubuntu-24.04`.
https://github.com/actions/runner-images/issues/10636

- In `ubuntu-24.04`, `R` is `not installed` by default
  A.`ubuntu-24.04`(`R` is `not installed` by default)
  https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2404-Readme.md#tools
  image

  B.`ubuntu-22.04`(`R` is `installed` by default)
  https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md#tools
  image

 

- Fix the failure issue of GA
  https://github.com/LuciferYang/spark/actions/runs/11268158324/job/31334445659
  image

### Does this PR introduce _any_ user-facing change?
No, only for tests.

### How was this patch tested?
Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: .github/workflows/build_and_test.yml
Comment (panbingkun) [inline .github/workflows/build_and_test.yml:1077]: https://github.com/actions/runner-images/issues/10636
image
Because the GitHub community is still in the process of `migration` (sometimes it's `ubuntu-22.04`, sometimes it's `ubuntu-22.04`), adding this is to verify the issue. After obtaining `approve`, I will remove it.

File: resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/BasicTestsSuite.scala
Comment (panbingkun) [inline resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/BasicTestsSuite.scala:33]: Just to trigger `k8s-integration-tests`, after obtaining `approve`, I will remove it.

**Comment by panbingkun** [comment]
cc @LuciferYang @dongjoon-hyun @HyukjinKwon

**Review by LuciferYang — APPROVED** [review]
+1, LGTM (pending tests)

Thanks @panbingkun , this solved the issue I encountered recently.

File: .github/workflows/build_and_test.yml
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:1077]: I think you should fix https://github.com/apache/spark/blob/master/dev/infra/Dockerfile?

Comment (LuciferYang) [reply .github/workflows/build_and_test.yml:1077]: https://github.com/apache/spark/blob/38d66fde2eec7e0d7bc9dda4d87cdee03b1ee9e8/dev/infra/Dockerfile#L61

```
 r-base \
```

This Docker file already has `r-base` installed.

Comment (panbingkun) [reply .github/workflows/build_and_test.yml:1077]: Yes, currently it is not running in a container
https://github.com/apache/spark/blob/38d66fde2eec7e0d7bc9dda4d87cdee03b1ee9e8/.github/workflows/build_and_test.yml#L1073-L1078

Comment (LuciferYang) [reply .github/workflows/build_and_test.yml:1077]: Perhaps we could try moving the tests into the container, and then install minikube by default, but only start it when running `k8s-integration-tests`? 

However, this configuration will be reused for the daily tests of branch-3.x, so we need to carefully consider how to ensure compatibility.

Comment (HyukjinKwon) [reply .github/workflows/build_and_test.yml:1077]: ah okok

Comment (HyukjinKwon) [reply .github/workflows/build_and_test.yml:1077]: I misread it. LG

Comment (LuciferYang) [reply .github/workflows/build_and_test.yml:1077]: haha

**Review by HyukjinKwon — APPROVED** [review]

**Comment by LuciferYang** [comment]
Merged into master. Thanks @panbingkun @HyukjinKwon

**Pull request closed without merging**
